src/components/AreaChart.tsx

```diff
@@ -0,0 +1,80 @@
+import * as React from 'react';
+import { omit } from 'lodash';
+import { Subtract } from 'utility-types';
+import { VisualizationObject, AFM } from '@gooddata/typings';
+
+import { AreaChart as AfmAreaChart } from './afm/AreaChart';
+
+import { ICommonChartProps } from './core/base/BaseChart';
+import { convertBucketsToAFM } from '../helpers/conversion';
+import { generateStackedDimensions } from '../helpers/dimensions';
+import { isStackedChart } from '../helpers/stacks';
+
+export interface IAreaChartBucketProps extends ICommonChartProps {
+    measures: VisualizationObject.BucketItem[];
+    viewBy?: VisualizationObject.IVisualizationAttribute[];
+    stackBy?: VisualizationObject.IVisualizationAttribute[];
+    filters?: VisualizationObject.VisualizationObjectFilter[];
+}
+
+export interface IAreaChartProps extends ICommonChartProps, IAreaChartBucketProps {
+    projectId: string;
+}
+
+type IAreaChartNonBucketProps = Subtract<IAreaChartProps, IAreaChartBucketProps>;
+
+export interface IAreaChartProps extends ICommonChartProps {
+    projectId: string;
+}
+
+function generateDefaultDimensions(afm: AFM.IAfm): AFM.IDimension[] {
+    return [
+        {
+            itemIdentifiers: ['measureGroup']
+        },
+        {
+            itemIdentifiers: (afm.attributes || []).map(a => a.localIdentifier)
```

Comment:
I would rewrite it as `get(afm, 'attributes', []).map(a => get(a, 'localIdentifier')